### PR TITLE
OHRM - Registration: with code refactoring

### PR DESCRIPTION
**Pull request by MadhukaHarith92**

### CODE REVIEW

File: symfony/plugins/orangehrmPimPlugin/lib/service/EmployeeService.php [hunk 1]
```diff
@@ -42,7 +42,11 @@ class EmployeeService extends BaseService {
      */
     private $employeeDao;
     private $configurationService;
+    private $orangeHrmRegistrationService;
 
+    const EMPLOYEE_COUNT_CHANGE = 10;
+    const EMPLOYEE_COUNT_INCREASE = 1;
```
Comment (rimaz523) [inline symfony/plugins/orangehrmPimPlugin/lib/service/EmployeeService.php]: Rename Constants as "EMPLOYEE_ACTIVE_COUNT" & "EMPLOYEE_INACTIVE_COUNT"

File: installer/OrangeHrmRegistration.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+class OrangeHrmRegistration
+{
+    /**
+     * Send the registration data captured during the installation
+     * @return bool
+     */
+    public function sendRegistrationData() {
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, "https://osreg-test-iris.orangehrm.com/registerAcceptor.php");
```
Comment (samanthajayasinghe) [inline installer/OrangeHrmRegistration.php]: can we move this static url to even class property now

File: installer/cli_install.php [hunk 3]
```diff
@@ -118,6 +124,26 @@ function getBasicConfigurations(){
...
+
+			$_SESSION['defUser']['organizationName'] = $detailsHandler->getOrganizationName();
+			$_SESSION['defUser']['adminEmployeeFirstName'] = $detailsHandler->getAdminEmployeeFirstName();
+			$_SESSION['defUser']['adminEmployeeLastName'] = $detailsHandler->getAdminEmployeeLastName();
+			$_SESSION['defUser']['organizationEmailAddress'] = $detailsHandler->getOrganizationEmailAddress();
+			$_SESSION['defUser']['contactNumber'] = $detailsHandler->getContactNumber();
+
+			$ohrmRegistration->sendRegistrationData();
+
+				$_SESSION['dbHostName'] = $detailsHandler->getHost();
```
Comment (samanthajayasinghe) [inline installer/cli_install.php]: align this line with other codes.

File: installer/defaultUser.php [hunk 4]
```diff
@@ -17,59 +17,254 @@
...
+    <form id="test" class="registrationFormBody" novalidate="false"></form>
+
+    <p>Please fill in your Organization details and Admin Username and Password for the Administrator login. </p>
+
+    <form id="registrationForm" name="registrationForm" novalidate="false" method="post" action="../install.php">
+        <h3>Instance Creation</h3>
+        <input type="hidden" id="actionResponse" name="actionResponse" value=""/>
+        <input type="hidden" id="type" name="type" value="0"/>
+        <label>Organization Name <em>*</em></label>
+        <input type="text" class="organizationName" style="margin-left: 55px; margin-bottom: 8px; padding-right: 130px" id="organizationName" name="organizationName"><br>
```
Comment (samanthajayasinghe) [inline installer/defaultUser.php]: still inline style here

File: symfony/plugins/orangehrmPimPlugin/lib/service/OrangeHrmRegisterService.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    private $configService;
+
+    /**
+     * Send details of employee count change to the server
+     * @return bool
+     * @throws CoreServiceException
+     */
+    public function sendRegistrationData() {
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, "https://osreg-test-iris.orangehrm.com/registerAcceptor.php");
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmPimPlugin/lib/service/OrangeHrmRegisterService.php]: better to add this url to config file and use in both classes

File: upgrader/lib/UpgradeOrangehrmRegistration.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+class UpgradeOrangehrmRegistration
+{
+    /**
+     * Send the registration data captured during the installation
+     * @return bool
+     */
+    public function sendRegistrationData() {
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, "https://osreg-test-iris.orangehrm.com/registerAcceptor.php");
```
Comment (samanthajayasinghe) [inline upgrader/lib/UpgradeOrangehrmRegistration.php]: same here

File: installer/OrangeHrmRegistration.php (hunk 2 above)
Comment (MadhukaHarith92) [reply installer/OrangeHrmRegistration.php]: Discussed with Rimaz and decided to keep it unchanged for the moment.

File: symfony/plugins/orangehrmPimPlugin/lib/service/OrangeHrmRegisterService.php (hunk 5 above)
Comment (MadhukaHarith92) [reply symfony/plugins/orangehrmPimPlugin/lib/service/OrangeHrmRegisterService.php]: Discussed with Rimaz and decided to keep it unchanged for the moment.

File: upgrader/lib/UpgradeOrangehrmRegistration.php (hunk 6 above)
Comment (MadhukaHarith92) [reply upgrader/lib/UpgradeOrangehrmRegistration.php]: Discussed with Rimaz and decided to keep it unchanged for the moment.

**Pull request merged**